### PR TITLE
Generated dialyzer types

### PR DESCRIPTION
**Pull request by mbuhot**
This PR enhances the `@type` attributes generated from the `command` macro.

Now each field is included with a type in the struct type.

### CODE REVIEW

File: lib/scout/commands/command.ex [hunk 1]
```diff
@@ -82,7 +82,7 @@ defmodule Scout.Commands.Command do
     [field_names: field_names, validation_asts: validation_funcs]
   ) do
     {
-      quote do {} end,
```
Comment (mbuhot) [inline lib/scout/commands/command.ex]: Changed this to empty quote block (which translates to `nil`) to make it easier to filter out.

File: lib/scout/commands/command.ex [hunk 2]
```diff
@@ -118,7 +118,46 @@ defmodule Scout.Commands.Command do
       [field_names: [name | field_names], validation_asts: [validation_func_ast | validation_funcs]]
     }
   end
-  defp attribute_visitor(node, accumulator), do: {node, accumulator}
+  defp attribute_visitor({:__block__, meta, fields}, accumulator) do
+    fields = Enum.filter(fields, &!is_nil(&1))
```
Comment (mbuhot) [inline lib/scout/commands/command.ex]: Kill the `nil` introduced by the standalone `validate` nodes once the traversal makes it to the root block node.

File: lib/scout/commands/command.ex [hunk 3]
```diff
@@ -118,7 +118,46 @@ defmodule Scout.Commands.Command do
...
+      :utc_datetime -> quote do DateTime.t end
+    end
+  end
+
+  defp build_type_ast(fields_ast = {:__block__, _, _}, module) do
+    Macro.postwalk(fields_ast, fn
+      {:field, _, [name, type | _rest]} ->  quote do {unquote(name),  unquote(ecto_to_typespec(type))} end
+      {:embeds_one, _, [name, type | _rest]} -> quote do {unquote(name), unquote(type).t} end
+      {:embeds_many, _, [name, type | _rest]} -> quote do {unquote(name), [unquote(type).t]} end
+      {:__block__, meta, args} -> {:%, [], [module, {:%{}, meta, args}]}
```
Comment (mbuhot) [inline lib/scout/commands/command.ex]: TIL this is what a struct constructor call looks like in AST.

File: lib/scout/commands/command.ex (hunk 2 above)
Comment (mattfurness) [reply lib/scout/commands/command.ex]: I'm going to have a think about negating the need for this

**Pull request merged**